crates/tools/js/benchmark/index.js

```diff
@@ -0,0 +1,217 @@
+  let totalFailures = 0;
+  for (let scenarioFile of scenarioFiles) {
+    if (grep && !scenarioFile.includes(grep)) {
+      continue;
+    }
+    // Get the filename from the path
+    const scenarioResult = await runScenario(
+      path.join(scenariosDir, scenarioFile)
+    );
+    totalTime += scenarioResult.result.timeMs;
+    totalFailures += scenarioResult.result.failures.length;
+    result[scenarioResult.name] = scenarioResult.result;
+  }
+
+  console.log(JSON.stringify(result));
+
+  // Log info to stderr so that it doesn't pollute stdout where we write the result
+  console.error(
+    `Total time ${
+      Math.round(100 * (totalTime / 1000)) / 100
+    } seconds with ${totalFailures} failures.`
+  );
+
+  process.exit(0);
+}
+
+async function runScenario(scenarioPath) {
+  const { config, requests } = await loadScenario(scenarioPath);
+  const name = path.basename(scenarioPath).split(".")[0];
+  console.error(`Running ${name} scenario`);
+
+  const start = performance.now();
+
+  const provider = await createHardhatNetworkProvider(config.providerConfig, {
+    enabled: config.loggerEnabled,
+  });
+
+  const failures = [];
+
+  for (let i = 0; i < requests.length; i += 1) {
+    try {
+      await provider.request(requests[i]);
+    } catch (e) {
+      failures.push(i);
+    }
+  }
+
+  const timeMs = performance.now() - start;
+
+  console.error(
+    `${name} finished in ${
+      Math.round(100 * (timeMs / 1000)) / 100
+    } seconds with ${failures.length} failures.`
+  );
+
+  return {
+    name,
+    result: {
+      timeMs,
+      failures,
+    },
+  };
+}
+
+async function loadScenario(path) {
+  const result = {
+    requests: [],
+  };
+  let i = 0;
+  for await (const line of readFile(path)) {
+    const parsed = JSON.parse(line);
+    if (i === 0) {
+      result.config = preprocessConfig(parsed);
+    } else {
+      result.requests.push(parsed);
+    }
+    i += 1;
+  }
+  return result;
+}
+
+function preprocessConfig(config) {
+  // From https://stackoverflow.com/a/59771233
+  const camelize = (obj) =>
+    _.transform(obj, (acc, value, key, target) => {
+      const camelKey = _.isArray(target) ? key : _.camelCase(key);
+
+      acc[camelKey] = _.isObject(value) ? camelize(value) : value;
+    });
+  config = camelize(config);
+
+  // EDR serializes None as null to json, but Hardhat expects it to be undefined
+  const removeNull = (obj) =>
+    _.transform(obj, (acc, value, key) => {
+      if (_.isObject(value)) {
+        acc[key] = removeNull(value);
+      } else if (!_.isNull(value)) {
+        acc[key] = value;
+      }
+    });
+  config = removeNull(config);
+
+  config.providerConfig.initialDate = new Date(
+    config.providerConfig.initialDate.secsSinceEpoch * 1000
+  );
+
+  config.providerConfig.hardfork = normalizeHardfork(
+    config.providerConfig.hardfork
+  );
+
+  // "accounts" in EDR are "genesisAccounts" in Hardhat
+  if (Object.keys(config.providerConfig.genesisAccounts).length !== 0) {
```

Comment:
The way I'm understanding this from looking at the code is that EDR has a concept of `genesis_accounts`, which is an initial list of impersonated accounts, and it also has an `accounts` field which is closer to what Hardhat calls `genesisAccounts`. Is this correct?

Reply:
Yes, the difference in EDR is that you need a secret key for accounts, but not for genesis accounts.